doc/pitfalls/data_leakage.rst

```diff
@@ -0,0 +1,152 @@
+.. _data_leakage:
+
+============
+Data leakage
+============
+
+Data leakage occurs when information that would not be available at prediction
+time, is used when building the model. This results in overly optimsitic
+performance estimates, for example from :ref:`cross validation
+<cross_validation>`, but poorer performance when the model is used
+on actually novel data, for example during production.
+
+A common cause is not keeping the test and train data subsets separate. Test
```

Comment:
I would write something like:

> The general rule is to never call `fit` on some testing data. While this may sound obvious, this is easy to miss in some cases, for example when applying some pre-processing steps.